### PR TITLE
OCD-3577: Update Surveillance reports

### PR DESCRIPTION
**Pull request by kekey1**
Sorry that this is going to be awful to review. The surveillance report code was scattered throughout and I put all its stuff under gov.healthit.chpl.surveillance.report packages. Many of the changes are due to that + removing Impl classes.

### CODE REVIEW

File: chpl/chpl-api/src/main/resources/log4j2-xinclude-loggers.xml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    </Logger>
+    <Logger name="quarterlyReportGenerationJobLogger" level="INFO"
+        additivity="false">
+        <AppenderRef ref="quarterlyReportGenerationJob" />
+    </Logger>
+    <Logger name="annualReportGenerationJobLogger" level="INFO"
+        additivity="false">
+        <AppenderRef ref="annualReportGenerationJob" />
+    </Logger>
+</Loggers>
```
Comment (andlar) [inline chpl/chpl-api/src/main/resources/log4j2-xinclude-loggers.xml]: The auto-checker called this one out. Not blocking approval, but it ought to be updated at some point

File: chpl/chpl-resources/src/main/resources/jobs.xml [hunk 2]
```diff
@@ -435,6 +435,46 @@
...
+            <name>annualReportGenerationJob</name>
+            <group>chplBackgroundJobs</group>
+            <description>Complete the work of generating an annual report for surveillance and sending it via email.</description>
+            <job-class>gov.healthit.chpl.scheduler.job.surveillanceReport.AnnualReportGenerationJob</job-class>
+            <durability>true</durability>
+            <recover>true</recover>
+            <job-data-map>
+                <entry>
+                    <key>authorities</key>
+                    <value>ROLE_ADMIN;ROLE_ONC;ROLE_ACB</value>
```
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/jobs.xml]: Should this be available to ROLE_ONC_STAFF? Maybe a question for the PO. Could also let it sit for now and update when we get there, but maybe worth creating it in the final state so we don't have to drop/recreate it again later

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/surveillanceReport/QuarterlyReportGenerationJob.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        try {
+            writtenFile = File.createTempFile(filename, ".xlsx");
+            outputStream = new FileOutputStream(writtenFile);
+            LOGGER.info("Writing quarterly report file to " + writtenFile.getAbsolutePath());
+            workbook.write(outputStream);
+        } catch (Exception ex) {
+            String msg = msgUtil.getMessage("report.quarterlySurveillance.export.writeError");
+            LOGGER.error(msg, ex);
+        } finally {
+            try { outputStream.flush(); } catch (Exception ignore) {}
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/surveillanceReport/QuarterlyReportGenerationJob.java]: There are some other checkstyle issues here. I think we want to touch on them, or figure out how to disable them if we don't agree with the rules, but one way or another I think we should work to not have errors reported before merging it

**Review by andlar — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/dto/QuarterlyReportExclusionDTO.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+@Data
+public class QuarterlyReportExclusionDTO {
+
+    private Long id;
+    private Long quarterlyReportId;
+    private Long listingId;
+    private String reason;
+
+    public QuarterlyReportExclusionDTO() {
+    }
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/dto/QuarterlyReportExclusionDTO.java]: Possible `@NoArgsConstructor`?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/dto/SurveillanceOutcomeDTO.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import gov.healthit.chpl.surveillance.report.entity.SurveillanceOutcomeEntity;
+import lombok.Data;
+
+@Data
+public class SurveillanceOutcomeDTO {
+
+    private Long id;
+    private String name;
+
+    public SurveillanceOutcomeDTO() {}
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/dto/SurveillanceOutcomeDTO.java]: `@NoArgsConstructor`?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/dto/SurveillanceProcessTypeDTO.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import gov.healthit.chpl.surveillance.report.entity.SurveillanceProcessTypeEntity;
+import lombok.Data;
+
+@Data
+public class SurveillanceProcessTypeDTO {
+
+    private Long id;
+    private String name;
+
+    public SurveillanceProcessTypeDTO() {}
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/dto/SurveillanceProcessTypeDTO.java]: `@NoArgsConstructor`?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/entity/QuarterEntity.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Column(name = "last_modified_user")
+    private Long lastModifiedUser;
+
+    @Column(name = "creation_date", insertable = false, updatable = false)
+    private Date creationDate;
+
+    @Column(name = "last_modified_date", insertable = false, updatable = false)
+    private Date lastModifiedDate;
+}
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/surveillance/report/entity/QuarterEntity.java]: Needs an `eol`?

**Review by tmy1313 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by tmy1313 — APPROVED** [review]

**Pull request merged**